src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTripTest.java

```diff
@@ -730,6 +732,114 @@ void canMakeOTPRequestAndResolveNoLongerPossibleTrip() throws Exception {
+        Leg walkLeg = itinerary.legs.get(0);
+        Leg transitLeg = itinerary.legs.get(1);
+        Leg finalLeg = itinerary.legs.get(2);
+
+        walkLeg.startTime = Date.from(walkLeg.startTime.toInstant().plusSeconds(departureDelay));
+        walkLeg.endTime = Date.from(walkLeg.endTime.toInstant().plusSeconds(departureDelay));
+
+        transitLeg.realTime = realTime;
+        transitLeg.departureDelay += departureDelay;
+        transitLeg.startTime = Date.from(transitLeg.startTime.toInstant().plusSeconds(departureDelay));
+        transitLeg.arrivalDelay += arrivalDelay;
+        transitLeg.endTime = Date.from(transitLeg.endTime.toInstant().plusSeconds(arrivalDelay));
+
+        finalLeg.startTime = Date.from(finalLeg.startTime.toInstant().plusSeconds(arrivalDelay));
+        finalLeg.endTime = Date.from(finalLeg.endTime.toInstant().plusSeconds(arrivalDelay));
+
+        itinerary.startTime = Date.from(itinerary.startTime.toInstant().plusSeconds(departureDelay));
+        itinerary.endTime = Date.from(itinerary.endTime.toInstant().plusSeconds(arrivalDelay));
+    }
+
+    /**
+     * A trip delay notification should be sent when saving a trip that has delays,
+     * and it is possible to find a matching trip without delays with the trip time in the OTP query params.
+     */
+    @Test
+    void canSendDelayNotificationOnSavingDelayedTrip() throws Exception {
+        OtpResponse mockWeekdayResponse = mockOtpPlanResponse();
+        // Add some delays in the transit leg, and remove any alerts.
+        Itinerary firstMockItinerary = firstItinerary(mockWeekdayResponse);
+        firstMockItinerary.clearAlerts();
+        addTransitLegDelay(firstMockItinerary, 300, 420, true);
+
+        // Create a mock monitored trip and CheckMonitorTrip instance
+        // Note that the response below gets modified from the original mockOtpPlanResponse.
+        CheckMonitoredTrip mockCheckMonitoredTrip = createCheckMonitoredTrip(() -> mockWeekdayResponse);
+        // Override matching itinerary to null to simulate initial save.
+        mockCheckMonitoredTrip.matchingItinerary = null;
+        MonitoredTrip mockTrip = mockCheckMonitoredTrip.trip;
+        // Trigger notifications for 5-minute delays instead of 15.
+        mockTrip.departureVarianceMinutesThreshold = 5;
+        mockTrip.arrivalVarianceMinutesThreshold = 5;
+        // Add delays to the original saved trip, different from the mock response.
+        // The delays from the mock response should be used.
+        addTransitLegDelay(mockTrip.itinerary, 600, 720, true);
+
+        Persistence.monitoredTrips.create(mockTrip);
+
+        // create mock itinerary existence for trip for Tuesdays
+        mockTrip.itineraryExistence.tuesday = new ItineraryExistence.ItineraryExistenceResult();
+
+        DateTimeUtils.useFixedClockAt(TUESDAY_20200609.withHour(8));
+
+        mockCheckMonitoredTrip.run();
```

Comment:
This might need help, should I loop reusing the `CheckMonitoredTrip` or create a new one and use a parameterized test?